crates/hotshot/task-impls/src/quorum_proposal_recv/handlers.rs

```diff
@@ -242,6 +243,157 @@ async fn validate_block_height<TYPES: NodeType>(
     Ok(())
 }
 
+async fn view_change_from_proposal<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions>(
```

Comment:
Maybe this is minor but IMO the logic should be structured as take the view from the VCE if it exists, else take it from the justify QC.  There is no case where VCE view is <= justify QC view.  I think this plus changing the syntax will make this function more concise and easy to read